c10/core/StorageImpl.cpp

```diff
@@ -1 +1,62 @@
 #include <c10/core/StorageImpl.h>
+
+namespace c10 {
+
+StorageImpl::StorageImpl(
+    caffe2::TypeMeta data_type,
+    int64_t numel,
+    at::DataPtr data_ptr,
+    at::Allocator* allocator,
+    bool resizable)
+    : data_type_(data_type),
+      data_ptr_(std::move(data_ptr)),
+      numel_(numel),
+      resizable_(resizable),
+      allocator_(allocator),
+      received_cuda_(false) {
+  if (numel > 0) {
+    if (data_type_.id() == caffe2::TypeIdentifier::uninitialized()) {
+      AT_ERROR(
+          "Constructing a storage with meta of unknown type and non-zero numel");
+    }
+  }
+}
+
+void StorageImpl::reset() {
+  data_ptr_.clear();
+  numel_ = 0;
+}
+
+StorageImpl::~StorageImpl() {}
```

Comment:
Nit: The small methods, my preference is to keep them in the header. They're unlikely to change, and it's nice to get them inlined ;) (So, concretely, these are reset, destructor, `release_resources`, `set_data_ptr`)

Reply:
(Maybe even the constructor, though that's more marginal)

Reply:
It's a little hard to tell; were there any substantive changes to these methods that got moved?

Reply:
Moved them temporarily as I had log messages inside them.

Reply:
Yeah, in general, it's easier for reviewers if there aren't reformatting changes; we have to wade through them when we review diffs and it makes it harder to spot substantive changes.